<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>846</width>
    <height>870</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QSerial</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_23" stretch="0,0">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>220</width>
          <height>300</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>220</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="title">
         <string>Serial Port Settings</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_7">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Port:</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QComboBox" name="serialPortComboBox"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Baud Rate:</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QComboBox" name="baudRateComboBox">
             <property name="editable">
              <bool>true</bool>
             </property>
             <property name="currentIndex">
              <number>7</number>
             </property>
             <item>
              <property name="text">
               <string>1200</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>2400</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>4800</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>9600</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>19200</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>38400</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>57600</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>115200</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>230400</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Data Bits:</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QComboBox" name="dataBitsComboBox">
             <property name="currentIndex">
              <number>3</number>
             </property>
             <item>
              <property name="text">
               <string>5</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>6</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>7</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>8</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Parity:</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_4">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QComboBox" name="parityComboBox">
             <item>
              <property name="text">
               <string>No</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Even</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Odd</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Space</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Mark</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Stop Bits:</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_5">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QComboBox" name="stopBitsComboBox">
             <item>
              <property name="text">
               <string>1</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>2</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>1.5</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <item>
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>Flow Control:</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_6">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QComboBox" name="flowControlComboBox">
             <item>
              <property name="text">
               <string>No</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Hardware</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Software</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer_3">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>220</width>
          <height>130</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>220</width>
          <height>130</height>
         </size>
        </property>
        <property name="title">
         <string>Recv Settings</string>
        </property>
        <property name="flat">
         <bool>false</bool>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_15">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout" stretch="1,1">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_7">
             <item>
              <widget class="QLabel" name="label_7">
               <property name="text">
                <string>Show As:</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_7">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QComboBox" name="recvShowAsComboBox">
               <item>
                <property name="text">
                 <string>Text (UTF-8)</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Text (Big5)</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Text (GB18030)</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Text (Shift-JIS)</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Hex</string>
                </property>
               </item>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_13">
             <item>
              <widget class="QCheckBox" name="recvShowTimeCheckBox">
               <property name="text">
                <string>Show Time</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pushButton">
               <property name="text">
                <string>Clear</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_3">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>220</width>
          <height>200</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>220</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="title">
         <string>Send Settings</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_6">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_9">
             <item>
              <widget class="QLabel" name="label_9">
               <property name="text">
                <string>Parse As:</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_9">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QComboBox" name="sendParseAsComboBox">
               <item>
                <property name="text">
                 <string>Text (UTF-8)</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Text (Big5)</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Text (GB18030)</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Text (Shift-JIS)</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Hex (No CRLF)</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Base64</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Percent Encoding</string>
                </property>
               </item>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_10">
             <item>
              <widget class="QLabel" name="label_11">
               <property name="text">
                <string>Line Ending:</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_10">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QComboBox" name="lineEndingComboBox">
               <item>
                <property name="text">
                 <string>LF (\n)</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>CR LF (\r \n)</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>CR (\r)</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>None</string>
                </property>
               </item>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_14">
             <item>
              <widget class="QCheckBox" name="sendShowTimeCheckBox">
               <property name="text">
                <string>Show Time</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="echoCheckBox">
               <property name="text">
                <string>Echo</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_12">
             <item>
              <widget class="QLabel" name="label_13">
               <property name="text">
                <string>Duration:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="breakDurationLineEdit">
               <property name="placeholderText">
                <string>ms</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="breakPushButton">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>BREAK</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QFrame" name="frame">
        <layout class="QHBoxLayout" name="horizontalLayout_16">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QPushButton" name="resetButton">
           <property name="text">
            <string>Reset Statistics</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>1</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="tab_text">
       <attribute name="title">
        <string>Text</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_18">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_4" stretch="3,1">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_11" stretch="4">
            <item>
             <widget class="QTextBrowser" name="textBrowser"/>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_8">
            <item>
             <widget class="QPlainTextEdit" name="inputPlainTextEdit">
              <property name="plainText">
               <string/>
              </property>
              <property name="placeholderText">
               <string>Shift+Enter to send</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="sendButton">
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>50</height>
               </size>
              </property>
              <property name="text">
               <string>Open and Send</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_term">
       <attribute name="title">
        <string>Terminal</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_17">
        <property name="leftMargin">
         <number>12</number>
        </property>
        <property name="rightMargin">
         <number>12</number>
        </property>
        <property name="bottomMargin">
         <number>12</number>
        </property>
        <item>
         <widget class="QWebEngineView" name="webEngineView" native="true">
          <property name="url" stdset="0">
           <url>
            <string>about:blank</string>
           </url>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>846</width>
     <height>37</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuSerial">
    <property name="title">
     <string>Serial</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionClose"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="actionMutual_Test"/>
   </widget>
   <addaction name="menuSerial"/>
   <addaction name="menuTools"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpen"/>
   <addaction name="actionClose"/>
   <addaction name="actionMutual_Test"/>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="toolTip">
    <string>Open serial port</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="text">
    <string>Close</string>
   </property>
   <property name="toolTip">
    <string>Close serial port</string>
   </property>
  </action>
  <action name="actionMutual_Test">
   <property name="text">
    <string>Mutual Test</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QWebEngineView</class>
   <extends>QWidget</extends>
   <header location="global">QtWebEngineWidgets/QWebEngineView</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>sendButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onSend()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>821</x>
     <y>639</y>
    </hint>
    <hint type="destinationlabel">
     <x>668</x>
     <y>594</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onOpen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>421</x>
     <y>380</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionClose</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onClose()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>421</x>
     <y>380</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>breakPushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onBreak()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>216</x>
     <y>713</y>
    </hint>
    <hint type="destinationlabel">
     <x>159</x>
     <y>725</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onClear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>176</x>
     <y>500</y>
    </hint>
    <hint type="destinationlabel">
     <x>226</x>
     <y>729</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionMutual_Test</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onMutualTest()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>421</x>
     <y>380</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>resetButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onReset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>122</x>
     <y>813</y>
    </hint>
    <hint type="destinationlabel">
     <x>422</x>
     <y>434</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onSend()</slot>
  <slot>onReset()</slot>
  <slot>onOpen()</slot>
  <slot>onClose()</slot>
  <slot>onBreak()</slot>
  <slot>onClear()</slot>
  <slot>onMutualTest()</slot>
 </slots>
</ui>
